pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchClient.java

```diff
@@ -73,9 +73,16 @@ public void afterBulk(long executionId, List<BulkProcessor.BulkOperationRequest>
                 }
                 int index = 0;
                 for (BulkProcessor.BulkOperationResult result: results) {
-                    final Record record = bulkOperationList.get(index++).getPulsarRecord();
+                    final Record<?> record = bulkOperationList.get(index++).getPulsarRecord();
                     if (result.isError()) {
                         record.fail();
+
+                        log.warn("Bulk request id={} failed, message id=[{}] index={} error={}", executionId,
```

Comment:
method `checkForIrrecoverableError` will log, ignore or throw an error for a malformed message error depending on `malformedDocAction`.
It'd be better to move this new log inside that method and ensure to log only if it's not a `MALFORMED_ERRORS`

Reply:
That makes sense. Please take another look.